"Unspecified error Waiting for Edge operation to terminate timed out"

Windows verification build failed on https://github.com/eclipse-platform/eclipse.platform/actions/runs/13118265699/job/36597841318?pr=1720
```
Error: Exception in thread "WorkbenchTestable" org.eclipse.swt.SWTException: Failed to execute runnable (org.eclipse.swt.SWTError: Unspecified error Waiting for Edge operation to terminate timed out)
	at org.eclipse.swt.SWT.error(SWT.java:4932)
	at org.eclipse.swt.SWT.error(SWT.java:4847)
	at org.eclipse.swt.widgets.Synchronizer.syncExec(Synchronizer.java:209)
	at org.eclipse.ui.internal.UISynchronizer.syncExec(UISynchronizer.java:133)
	at org.eclipse.swt.widgets.Display.syncExec(Display.java:4836)
	at org.eclipse.e4.ui.internal.workbench.swt.E4Testable.testingFinished(E4Testable.java:157)
	at org.eclipse.tycho.surefire.osgibooter.AbstractUITestApplication.runTests(AbstractUITestApplication.java:57)
	at org.eclipse.e4.ui.internal.workbench.swt.E4Testable.lambda$0(E4Testable.java:79)
	at java.base/java.lang.Thread.run(Thread.java:1583)
Caused by: org.eclipse.swt.SWTError: Unspecified error Waiting for Edge operation to terminate timed out
	at org.eclipse.swt.SWT.error(SWT.java:4957)
	at org.eclipse.swt.browser.Edge.processOSMessagesUntil(Edge.java:474)
	at org.eclipse.swt.browser.Edge$WebViewProvider.getWebViewWrapper(Edge.java:370)
	at org.eclipse.swt.browser.Edge$WebViewProvider.getWebView(Edge.java:381)
	at org.eclipse.swt.browser.Edge.getUrl(Edge.java:893)
	at org.eclipse.swt.browser.Browser.getUrl(Browser.java:771)
	at org.eclipse.ui.internal.intro.impl.presentations.BrowserIntroPartImplementation.saveCurrentPage(BrowserIntroPartImplementation.java:645)
	at org.eclipse.ui.internal.intro.impl.model.AbstractIntroPartImplementation.saveState(AbstractIntroPartImplementation.java:302)
	at org.eclipse.ui.internal.intro.impl.model.IntroPartPresentation.saveState(IntroPartPresentation.java:427)
	at org.eclipse.ui.intro.config.CustomizableIntroPart.saveState(CustomizableIntroPart.java:405)
	at org.eclipse.ui.internal.ViewIntroAdapterPart.saveState(ViewIntroAdapterPart.java:198)
	at org.eclipse.ui.internal.ViewReference.persist(ViewReference.java:60)
	at org.eclipse.ui.internal.e4.compatibility.CompatibilityView.persistState(CompatibilityView.java:226)
	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
	at java.base/java.lang.reflect.Method.invoke(Method.java:580)
	at org.eclipse.e4.core.internal.di.MethodRequestor.execute(MethodRequestor.java:56)
	at org.eclipse.e4.core.internal.di.InjectorImpl.invokeUsingClass(InjectorImpl.java:299)
	at org.eclipse.e4.core.internal.di.InjectorImpl.invoke(InjectorImpl.java:227)
	at org.eclipse.e4.core.contexts.ContextInjectionFactory.invoke(ContextInjectionFactory.java:148)
	at org.eclipse.e4.ui.internal.workbench.swt.PartRenderingEngine.safeRemoveGui(PartRenderingEngine.java:925)
	at org.eclipse.e4.ui.internal.workbench.swt.PartRenderingEngine$4.run(PartRenderingEngine.java:857)
	at org.eclipse.core.runtime.SafeRunner.run(SafeRunner.java:47)
	at org.eclipse.e4.ui.internal.workbench.swt.PartRenderingEngine.removeGui(PartRenderingEngine.java:841)
	at org.eclipse.e4.ui.workbench.renderers.swt.ElementReferenceRenderer.disposeWidget(ElementReferenceRenderer.java:113)
	at org.eclipse.e4.ui.internal.workbench.swt.PartRenderingEngine.safeRemoveGui(PartRenderingEngine.java:934)
	at org.eclipse.e4.ui.internal.workbench.swt.PartRenderingEngine$4.run(PartRenderingEngine.java:857)
	at org.eclipse.core.runtime.SafeRunner.run(SafeRunner.java:47)
	at org.eclipse.e4.ui.internal.workbench.swt.PartRenderingEngine.removeGui(PartRenderingEngine.java:841)
	at org.eclipse.e4.ui.internal.workbench.swt.PartRenderingEngine.safeRemoveGui(PartRenderingEngine.java:896)
	at org.eclipse.e4.ui.internal.workbench.swt.PartRenderingEngine$4.run(PartRenderingEngine.java:857)
	at org.eclipse.core.runtime.SafeRunner.run(SafeRunner.java:47)
	at org.eclipse.e4.ui.internal.workbench.swt.PartRenderingEngine.removeGui(PartRenderingEngine.java:841)
	at org.eclipse.e4.ui.internal.workbench.swt.PartRenderingEngine.safeRemoveGui(PartRenderingEngine.java:896)
	at org.eclipse.e4.ui.internal.workbench.swt.PartRenderingEngine$4.run(PartRenderingEngine.java:857)
	at org.eclipse.core.runtime.SafeRunner.run(SafeRunner.java:47)
	at org.eclipse.e4.ui.internal.workbench.swt.PartRenderingEngine.removeGui(PartRenderingEngine.java:841)
	at org.eclipse.e4.ui.internal.workbench.swt.PartRenderingEngine.safeRemoveGui(PartRenderingEngine.java:896)
	at org.eclipse.e4.ui.internal.workbench.swt.PartRenderingEngine$4.run(PartRenderingEngine.java:857)
	at org.eclipse.core.runtime.SafeRunner.run(SafeRunner.java:47)
	at org.eclipse.e4.ui.internal.workbench.swt.PartRenderingEngine.removeGui(PartRenderingEngine.java:841)
	at org.eclipse.e4.ui.internal.workbench.swt.PartRenderingEngine.stop(PartRenderingEngine.java:1219)
	at org.eclipse.e4.ui.internal.workbench.E4Workbench.close(E4Workbench.java:190)
	at org.eclipse.e4.ui.internal.workbench.swt.E4Testable.lambda$3(E4Testable.java:157)
	at org.eclipse.swt.widgets.RunnableLock.run(RunnableLock.java:40)
	at org.eclipse.swt.widgets.Synchronizer.runAsyncMessages(Synchronizer.java:132)
	at org.eclipse.swt.widgets.Display.runAsyncMessages(Display.java:4077)
	at org.eclipse.swt.widgets.Display.readAndDispatch(Display.java:3693)
	at org.eclipse.e4.ui.internal.workbench.swt.PartRenderingEngine$5.run(PartRenderingEngine.java:1151)
	at org.eclipse.core.databinding.observable.Realm.runWithDefault(Realm.java:339)
	at org.eclipse.e4.ui.internal.workbench.swt.PartRenderingEngine.run(PartRenderingEngine.java:1042)
	at org.eclipse.e4.ui.internal.workbench.E4Workbench.createAndRunUI(E4Workbench.java:153)
	at org.eclipse.ui.internal.Workbench.lambda$3(Workbench.java:668)
	at org.eclipse.core.databinding.observable.Realm.runWithDefault(Realm.java:339)
	at org.eclipse.ui.internal.Workbench.createAndRunWorkbench(Workbench.java:576)
	at org.eclipse.ui.PlatformUI.createAndRunWorkbench(PlatformUI.java:173)
	at org.eclipse.ui.internal.ide.application.IDEApplication.start(IDEApplication.java:178)
	at org.eclipse.tycho.surefire.osgibooter.UITestApplication.runApplication(UITestApplication.java:34)
	at org.eclipse.tycho.surefire.osgibooter.AbstractUITestApplication.run(AbstractUITestApplication.java:129)
	at org.eclipse.tycho.surefire.osgibooter.UITestApplication.start(UITestApplication.java:44)
	at org.eclipse.equinox.internal.app.EclipseAppHandle.run(EclipseAppHandle.java:208)
	at org.eclipse.core.runtime.internal.adaptor.EclipseAppLauncher.runApplication(EclipseAppLauncher.java:149)
	at org.eclipse.core.runtime.internal.adaptor.EclipseAppLauncher.start(EclipseAppLauncher.java:115)
	at org.eclipse.core.runtime.adaptor.EclipseStarter.run(EclipseStarter.java:467)
	at org.eclipse.core.runtime.adaptor.EclipseStarter.run(EclipseStarter.java:298)
	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
	at java.base/java.lang.reflect.Method.invoke(Method.java:580)
	at org.eclipse.equinox.launcher.Main.invokeFramework(Main.java:668)
	at org.eclipse.equinox.launcher.Main.basicRun(Main.java:605)
	at org.eclipse.equinox.launcher.Main.run(Main.java:1481)
	at org.eclipse.equinox.launcher.Main.main(Main.java:1454)
Error: The operation was canceled.
```

--- a/bundles/org.eclipse.e4.ui.bindings/META-INF/MANIFEST.MF
+++ b/bundles/org.eclipse.e4.ui.bindings/META-INF/MANIFEST.MF
@@ -1,7 +1,7 @@
 Manifest-Version: 1.0
 Bundle-ManifestVersion: 2
 Bundle-SymbolicName: org.eclipse.e4.ui.bindings;singleton:=true
-Bundle-Version: 0.15.0.qualifier
+Bundle-Version: 0.15.100.qualifier
 Bundle-Name: %pluginName
 Bundle-Vendor: %providerName
 Bundle-Localization: plugin

--- a/bundles/org.eclipse.e4.ui.bindings/src/org/eclipse/e4/ui/bindings/keys/KeyBindingDispatcher.java
+++ b/bundles/org.eclipse.e4.ui.bindings/src/org/eclipse/e4/ui/bindings/keys/KeyBindingDispatcher.java
@@ -295,32 +295,42 @@ public class KeyBindingDispatcher {
 		final EHandlerService handlerService = getHandlerService();
 		final Command command = parameterizedCommand.getCommand();
 
-		final IEclipseContext staticContext = createContext(trigger);
-
 		final boolean commandDefined = command.isDefined();
 		// boolean commandEnabled;
 		boolean commandHandled = false;
 
+		// commandEnabled = handlerService.canExecute(parameterizedCommand,
+		// staticContext);
+		Object obj = HandlerServiceImpl.lookUpHandler(context, command.getId());
+		if (obj != null) {
+			if (obj instanceof IHandler handler) {
+				commandHandled = command.isEnabled() && handler.isHandled();
+			} else {
+				commandHandled = true;
+			}
+		}
+
+		if (isTracingEnabled()) {
+			logger.trace("Command " + parameterizedCommand + ", defined: " + commandDefined + ", handled: " //$NON-NLS-1$ //$NON-NLS-2$ //$NON-NLS-3$
+					+ commandHandled + " in " + describe(context)); //$NON-NLS-1$
+		}
+
+		if (trigger != null && trigger.widget instanceof Browser) {
+			getDisplay()
+					.asyncExec(() -> handleCommandExecution(parameterizedCommand, handlerService, trigger, obj));
+		} else {
+			handleCommandExecution(parameterizedCommand, handlerService, trigger, obj);
+		}
+		return (commandDefined && commandHandled);
+	}
+
+	private void handleCommandExecution(final ParameterizedCommand parameterizedCommand,
+			final EHandlerService handlerService, final Event trigger, Object handler) {
+		final IEclipseContext staticContext = createContext(trigger);
 		try {
-			// commandEnabled = handlerService.canExecute(parameterizedCommand, staticContext);
-			Object obj = HandlerServiceImpl.lookUpHandler(context, command.getId());
-			if (obj != null) {
-				if (obj instanceof IHandler) {
-					commandHandled = ((IHandler) obj).isHandled();
-				} else {
-					commandHandled = true;
-				}
-			}
-
-			if (isTracingEnabled()) {
-				logger.trace("Command " + parameterizedCommand + ", defined: " + commandDefined + ", handled: " //$NON-NLS-1$ //$NON-NLS-2$ //$NON-NLS-3$
-						+ commandHandled + " in " + describe(context)); //$NON-NLS-1$
-			}
-
 			handlerService.executeHandler(parameterizedCommand, staticContext);
 			final Object commandException = staticContext.get(HandlerServiceImpl.HANDLER_EXCEPTION);
 			if (commandException instanceof CommandException) {
-				commandHandled = false;
 				if (commandException instanceof ExecutionException) {
 					if (logger != null) {
 						logger.error((Throwable) commandException,
@@ -328,8 +338,7 @@ public class KeyBindingDispatcher {
 										+ describe(context));
 					}
 				} else if (isTracingEnabled()) {
-					logger.trace((Throwable) commandException,
-							"Command exception for: " + parameterizedCommand + " in " //$NON-NLS-1$ //$NON-NLS-2$
+					logger.trace((Throwable) commandException, "Command exception for: " + parameterizedCommand + " in " //$NON-NLS-1$ //$NON-NLS-2$
 									+ describe(context));
 					if (handlerService instanceof HandlerServiceImpl serviceImpl) {
 						IEclipseContext serviceContext = serviceImpl.getContext();
@@ -337,7 +346,7 @@ public class KeyBindingDispatcher {
 							StringBuilder sb = new StringBuilder("\n\tExecution context: "); //$NON-NLS-1$
 							sb.append(describe(serviceContext));
 							sb.append("\n\tHandler: "); //$NON-NLS-1$
-							sb.append(obj);
+							sb.append(handler);
 							logger.trace(sb.toString());
 						}
 					}
@@ -352,17 +361,16 @@ public class KeyBindingDispatcher {
 					}
 				}
 			}
-			/*
-			 * Now that the command has executed (and had the opportunity to use the remembered
-			 * state of the dialog), it is safe to delete that information.
-			 */
-			if (keyAssistDialog != null) {
-				keyAssistDialog.clearRememberedState();
-			}
 		} finally {
 			staticContext.dispose();
 		}
-		return (commandDefined && commandHandled);
+		/*
+		 * Now that the command has executed (and had the opportunity to use the
+		 * remembered state of the dialog), it is safe to delete that information.
+		 */
+		if (keyAssistDialog != null) {
+			keyAssistDialog.clearRememberedState();
+		}
 	}
 
 	private boolean isTracingEnabled() {

